### PR TITLE
MAIN-4151 set memc key instead of reading it only

### PR DESCRIPTION
**Pull request by bognix**

### CODE REVIEW

File: extensions/wikia/GlobalFooter/GlobalFooterController.class.php [hunk 1]
```diff
@@ -80,6 +80,8 @@ private function getGlobalFooterLinks() {
 			}
 		}
 
+		$wgMemc->set($memcKey, $parsedLinks);
```
Comment (macbre) [inline extensions/wikia/GlobalFooter/GlobalFooterController.class.php]: Per our Slack discussion:

* add `$ttl` argument
* spacing :)

Comment (macbre) [inline extensions/wikia/GlobalFooter/GlobalFooterController.class.php]: Please add error reporting in line 66 if we treat such case as an error.

**Comment by macbre** [comment]
Jolly good! Thanks for fixing it so quick :)

File: extensions/wikia/GlobalFooter/GlobalFooterController.class.php [hunk 2]
```diff
@@ -80,6 +86,8 @@ private function getGlobalFooterLinks() {
 			}
 		}
 
+		$wgMemc->set( $memcKey, $parsedLinks, self::MEMC_EXPIRY );
```
Comment (Warkot) [inline extensions/wikia/GlobalFooter/GlobalFooterController.class.php]: There's a slight possibility that we cache an empty response i.e. when:
* there is no message in Memcache
* AND `getMessageAsArray()` finds a message, but its content is empty (`is_null($var)` is equal to `$var === NULL`, so an empty response is still not a `NULL`—lines 65-66).
But that may be also intended. :confused:

Comment (bognix) [inline extensions/wikia/GlobalFooter/GlobalFooterController.class.php]: I believe if message key exists we should display its value despite what it contains and if it contains anything. As you said that might be an error but on the other hand that may be intended. Messages are controlled by us so I believe we can assume if key exists then we should show whatever value it contains.

Comment (Warkot) [inline extensions/wikia/GlobalFooter/GlobalFooterController.class.php]: Agreed, let's keep it this way. Thanks for the reply!

**Comment by Warkot** [comment]
LGTM :shipit:

**Pull request merged**